### DISCUSSION TITLE
How to add a custom parition

### DISCUSSION THREAD

**Discussion by apbr** [OE4T/meta-tegra#1245] [Q&A] [answered]
Hi,

I'm, trying to add an additional partition to the Jetson image. This partition to be formatted (e.g. with ext4) and contain some files installed by Yocto recipes.
I build for the Jetson Orin AGX.

By looking at https://github.com/madisongh/tegra-test-distro/blob/master/layers/meta-testdistro/recipes-bsp/tegra-binaries/custom-flash-layout_1.0.bb I was able to create the partition but it is not formatted and I don't know how to create files.

Can anyone give my some hints on how to do this?

**Comment by ichergui** [comment]
Hello @apbr 
Welcome to the OE4T community.
That the right approach to introduce a custom flash xml file.
You need to create the image your self . 
You have to use this [function](https://github.com/OE4T/meta-tegra/blob/master/classes/image_types_tegra.bbclass#L120) for the new image creation
Let me know if you need a further details

**Reply by apbr** [reply]
Hello @ichergui,
thank you for your response.
> You need to create the image your self .

How would I create the image for this additional partition with Yocto?
(Sorry, I guess this is quite a basic question, but I've never touched this part of Yocto)

**Reply by dwalkes** [reply, marked as answer]
@apbr have you noticed https://github.com/OE4T/meta-tegra/discussions/858#discussioncomment-1664607 ?  This might be a good place to start.

**Reply by ichergui** [reply]
Thanks @dwalkes

**Reply by apbr** [reply]
Thanks @dwalkes this helps a lot. (No idea why I didn't find this by myself.)
I now managed to create an image with an additional formatted partition.

But I probably need to populate it with some files.
When using WIC (for some other non-Jetson boards) I could tell WIC to put everything except `/data` into the rootfs and create an additional partition from the contents of `/data`. Is there a similar mechanism for tegraflash?

**Reply by dwalkes** [reply]
> Is there a similar mechanism for tegraflash?

Not that I'm aware of.  You've probably found the [discussions about WIC](https://github.com/OE4T/meta-tegra/search?q=WIC).  Several people have asked about this but nobody has taken on any integration AFAIK.

**Reply by apbr** [reply]
Ok, thank you for all the help.
